src/pe/header.rs

```diff
@@ -792,15 +813,11 @@ pub struct Header {
     pub optional_header: Option<optional_header::OptionalHeader>,
 }
 
-impl Header {
-    pub fn parse(bytes: &[u8]) -> error::Result<Self> {
+impl<'a> Header<'a> {
```

Comment:
this is also technically a breaking change; i believe you message makes clear why this should now be borrowed/have a lifetime